ConsentViewController/Classes/Consents/SPPreferencesConsent.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  SPPreferencesConsent.swift
+//  Pods
+//
+//  Created by Fedko Dmytro on 06.05.2025
+//
+
+import Foundation
+
+@objcMembers public class SPPreferencesConsent: NSObject, Codable, CampaignConsent, NSCopying {
+    var applies: Bool = true
+    var dateCreated: SPDate
+    var messageId: String?
+    public var uuid: String?
+    public var status, rejectedStatus: [PreferencesStatus]?
+
+
+    init(
+        dateCreated: SPDate,
+        messageId: String? = nil,
+        uuid: String? = nil,
+        status: [PreferencesStatus]?,
+        rejectedStatus: [PreferencesStatus]?
```

Comment:
Can we add `nil` as default params for both `status` and `rejectedStatus`?  and do they really need to be optional? I think it's a good practice to try and avoid optionals unless it's really a necessity.